### PR TITLE
Adds RELEASE.md with documentation for releasing NuPIC.

### LINKED ISSUE #3346: Add release documentation to repo
Add a RELEASE.md file with instructions for how to release NuPIC.

### PR DESCRIPTION
**Pull request by scottpurdy**
fixes #3346 

@rhyolight please review

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Release Process
+
+1. Send announcement that a release is underway to the committer's lounge on
+discourse.numenta.org and ask reviews not to merge PRs in NuPIC until you're
```
Comment (rhyolight) [inline RELEASE.md]: "ask reviewers"

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+done with the release.
+2. Create a PR that includes:
+    - Release notes - these can be generated by Matt's tool (TODO: add link)
```
Comment (rhyolight) [inline RELEASE.md]: I will need to make changes to this tool before it can be used out of context.

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+3. Wait for the PR to be approved and merged, and the Bamboo build to complete
+successfully
+4. Create a "release" in Bamboo with a version matching the intended release
+version
+5. Deploy the release in Bamboo. This will:
+    - Validate that the Bamboo release number matches the wheel version
+    - Check that release notes are present for the version to be released
+    - Push the wheel to PyPI
+    - If successful, push a version tag to the repo
+6. Send announcement to the committer's lounge on discourse.numenta.org that the release is complete.
```
Comment (rhyolight) [inline RELEASE.md]: I think the announcement should be posted in the [NuPIC](https://discourse.numenta.org/c/nupic) category.

Comment (scottpurdy) [inline RELEASE.md]: I think the announcement needs to be in a low-volume channel that we can expect all reviewers (optionally all committers, or anyone else as well) to be actively monitoring. I would guess that some reviewers are not getting notifications, or are using digest or email filtering for the NuPIC category.

Comment (scottpurdy) [inline RELEASE.md]: @rhyolight perhaps we add a "NuPIC Announce" category that only reviewers can post to and that we use specifically for release or other important announcements?

Comment (rhyolight) [inline RELEASE.md]: Something like that sounds ok by me. We'll play it by ear.

File: RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+1. Send announcement that a release is underway to the committer's lounge on
+discourse.numenta.org and ask reviewers not to merge PRs in NuPIC until you're
+done with the release.
+2. Create a PR that includes:
+    - Release notes added to CHANGELOG.md
+    - Change to the VERSION file so it matches the intended release version
+3. Wait for the PR to be approved and merged, and the Bamboo build to complete
+successfully
+4. Create a "release" in Bamboo with a version matching the intended release
```
Comment (rhyolight) [inline RELEASE.md]: BTW, I think we can do away with this step at some point if we can figure out how to make the contents of `VERSION.txt` available as a [Bamboo deployment variable](https://confluence.atlassian.com/bamboo/variables-for-deployment-environments-342754180.html).

**Comment by rhyolight** [comment]
👍  with the knowledge that it will evolve over time.

File: RELEASE.md (hunk 4 above)
Comment (scottpurdy) [inline RELEASE.md]: You have to create a "release" before you can deploy. The version you specify there is an arbitrary string (with a simple mechanism that bumps the last number as the default next version) and that version does become available as a deployment variable. So we have to create the Bamboo release and I don't know of a way to enforce that the version matches something from the build so we have to validate that in the deployment.

**Pull request merged**